src/Microsoft.OData.Core/UriParser/Metadata/ODataUriResolver.cs

```diff
@@ -276,47 +276,40 @@ public virtual IEnumerable<IEdmOperationImport> ResolveOperationImports(IEdmMode
         /// <returns>The resolved key list.</returns>
         public virtual IEnumerable<KeyValuePair<string, object>> ResolveKeys(IEdmEntityType type, IDictionary<string, string> namedValues, Func<IEdmTypeReference, string, object> convertFunc)

```

Comment:
I will add conditional check and add call to resolve using alternate keys if resolve using primary key fails.